tests/test_sleep_lock_hygiene.py

```
@@ -153,3 +153,39 @@ def raising_embed(texts):
+    on that thread failed "database is locked" instantly until something
+    reset it. Observed in production as a self-healing storm of instant
+    tool failures minutes after the real writer had finished."""
+
+    def test_touch_failure_leaves_no_open_transaction(self, temp_db, monkeypatch):
+        beam = BeamMemory(session_id="touch-rollback", db_path=temp_db)
+        beam.remember("touch rollback seed", source="test")
+
+        real_commit = beam.conn.commit
+        calls = {"n": 0}
+
+        def failing_commit():
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `failing_commit`

(ANN202)